File: ter-semestre/dados-armazenamento/ed-01-exercicio-temas-diagrama/src/main/java/org/example/Funcionario.java
package org.example;

public abstract class Funcionario extends Pessoa {
    private double salario;

    public Funcionario(String nome, int cpf, double salario) {
        super(nome, cpf);
        this.salario = salario;
    }

    public double getSalario() {
        return salario;
    }

    public void setSalario(double salario) {
        this.salario = salario;
    }
}

File: kotlin/src/main/kotlin/Funcionario.kt
class Funcionario(
    nome: String,
    idade: Int,
    casado: Boolean? = null,
    altura: Double? = null,
    var salario: Double
) : Pessoa(nome, idade, casado, altura) {


    override fun toString(): String {
        return "Funcionario(salario=$salario) ${super.toString()}"
    }
}